akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiNodeClusterShardingSpec.scala

```diff
@@ -72,67 +105,112 @@ abstract class MultiNodeClusterShardingSpec(val config: MultiNodeClusterSharding
     super.afterTermination()
   }
 
-  protected def join(from: RoleName, to: RoleName): Unit = {
+  /**
+   * Flexible cluster join pattern usage.
+   *
+   * @param from the node the `Cluster.join` is `runOn`
+   * @param to the node to join to
+   * @param assert if disabled - false, the joining member's post-join status assertions
+   *               are not run. This allows tests that were not doing assertions or
+   *               doing them after `onJoinedRunOnFrom` to have more flexibility.
+   *               Defaults to true, and running member status checks.
+   * @param onJoinedRunOnFrom allows you to execute a function after join validation
+   *                          is successful, e.g. starting sharding
+   */
+  protected def join(from: RoleName, to: RoleName, onJoinedRunOnFrom: => Unit = Unit, assert: Boolean = true): Unit = {
     runOn(from) {
-      Cluster(system).join(node(to).address)
-      awaitAssert {
-        Cluster(system).state.isMemberUp(node(from).address)
+      cluster.join(node(to).address)
+      if (assert) {
+        within(20.seconds) {
+          awaitAssert {
+            // let's pick one or two vs have 3
```

Comment:
I don't understand that comment

Reply:
The comment is that some tests use this one assertion, some the other two, we should consolidate and pick the most appropriate if possible.

Reply:
👍 i think the first is fine, no need to check all the others are up

Reply:
LGTM thanks @chbatey